test/react_client/event_helpers_test.dart

```diff
@@ -1657,28 +1793,52 @@ main() {
           test('when the event is a different type', () {
             expect(eventTypeTester(createSyntheticClipboardEvent()),
                 currentEventTypeBeingTested == SyntheticEventType.syntheticClipboardEvent ? isTrue : isFalse);
-            expect(eventTypeTester(createSyntheticKeyboardEvent()),
-                currentEventTypeBeingTested == SyntheticEventType.syntheticKeyboardEvent ? isTrue : isFalse);
+            expect(
+                eventTypeTester(createSyntheticKeyboardEvent()),
+                (currentEventTypeBeingTested == SyntheticEventType.syntheticKeyboardEvent ||
+                        currentEventTypeBeingTested == SyntheticEventType.syntheticUIEvent)
+                    ? isTrue
+                    : isFalse);
```

Comment:
Changes like this are necessary since `SyntheticKeyboardEvent` extends from `SyntheticUIEvent`.